Makefile

```diff
@@ -0,0 +1,88 @@
+# TODO: run golint, errcheck, staticcheck, unused
+.PHONY: default
+default: deps checkgofmt vet predeclared test
+
+.PHONY: ci
```

Comment:
Why is this not the default? (The default is what is actually being run from CI.)

Maybe the TODO should be to remove the `default` target and use `ci` instead. And then you could add those other checks (staticcheck, unused, and ineffassign) here.

Reply:
For normal dev testing I left the default make target which can also be used for travis where codecov.io integration is not desired. Note I added -cover for the test target so devs can see coverage summary stats in console out before they push.

The ci target uses testcover instead of the test target. The testcover target outputs coverage.out for codecov.io integration. I thought you might want to start using codecov.io to help identify areas of the code that aren't being tested, so left a separate ci target you could use in your travis file like this: https://github.com/prasek/protoreflect/blob/grpc/.travis.yml.

If you want to start using codecov.io, or want travis to use make ci regardless,  I can push an updated travis file.

Reply:
Also, note that make testcover (~60s) is 5x slower vs make test (~12s). For normal dev testing before push, having the default target (using test) is definitely much nicer vs the ci target (using testcover).